lib/cisco_node_utils/interface_ospf.rb

```diff
@@ -143,109 +150,128 @@ def default_message_digest_encryption_type
+      config_get('interface_ospf', 'message_digest_password', @get_args)
     end
 
     def default_message_digest_password
       config_get_default('interface_ospf', 'message_digest_password')
     end
 
-    # interface %s
-    #   %s ip ospf message-digest-key %d %s %d %s
     def message_digest_key_set(keyid, algtype, enctype, enc)
       current_keyid = message_digest_key_id
       if keyid == default_message_digest_key_id && current_keyid != keyid
-        config_set('interface_ospf', 'message_digest_key_set',
-                   @interface.name, 'no', current_keyid,
-                   '', '', '')
+        @set_args.merge!(state:   'no',
+                         keyid:   current_keyid,
+                         algtype: '',
+                         enctype: '',
+                         enc:     '')
+        config_set('interface_ospf', 'message_digest_key_set', @set_args)
       elsif keyid != default_message_digest_key_id
         fail TypeError unless enc.is_a?(String)
         fail ArgumentError unless enc.length > 0
         enctype = Encryption.symbol_to_cli(enctype)
-        config_set('interface_ospf', 'message_digest_key_set',
-                   @interface.name, '', keyid, algtype, enctype, enc)
+        @set_args.merge!(state:   '',
+                         keyid:   current_keyid,
+                         algtype: algtype,
+                         enctype: enctype,
+                         enc:     enc)
+        config_set('interface_ospf', 'message_digest_key_set', @set_args)
       end
+      set_args_keys_default
     end
 
     def cost
-      config_get('interface_ospf', 'cost', @interface.name)
+      config_get('interface_ospf', 'cost', @get_args)
     end
 
     def default_cost
       config_get_default('interface_ospf', 'cost')
     end
 
-    # interface %s
-    #   ip ospf cost %d
     def cost=(c)
       if c == default_cost
-        config_set('interface_ospf', 'cost', @interface.name, 'no', '')
+        @set_args.merge!(state: 'no', cost: '')
       else
-        config_set('interface_ospf', 'cost', @interface.name, '', c)
+        @set_args.merge!(state: '', cost: c)
       end
+      config_set('interface_ospf', 'cost', @set_args)
+      set_args_keys_default
     end
 
     def hello_interval
-      config_get('interface_ospf', 'hello_interval', @interface.name)
+      config_get('interface_ospf', 'hello_interval', @get_args)
     end
 
     def default_hello_interval
       config_get_default('interface_ospf', 'hello_interval')
     end
 
-    # interface %s
-    #   ip ospf hello-interval %d
     def hello_interval=(interval)
-      config_set('interface_ospf', 'hello_interval',
-                 @interface.name, '', interval.to_i)
+      # Previous behavior always sets interval and ignores 'no' cmd
+      @set_args.merge!(state: '', interval: interval.to_i)
+      config_set('interface_ospf', 'hello_interval', @set_args)
+      set_args_keys_default
+    end
+
+    def destroy_hello_interval
```

Comment:
Minor but can you combine `destroy_hello_interval` and `destroy_dead_interval` into one helper method instead of two?  Code is identical except for `dead_interval`|`hello_interval` string.

Reply:
done